drivers/digital-io/max22190/max22190.c

```diff
@@ -0,0 +1,341 @@
+	int ret;
+
+	if (desc->crc_en)
+		xfer.bytes_number++;
+
+	memset(desc->buff, 0, xfer.bytes_number);
+	desc->buff[0] = no_os_field_prep(MAX22190_ADDR_MASK, addr) |
+			no_os_field_prep(MAX22190_RW_MASK, 0);
+
+	if (desc->crc_en)
+		desc->buff[2] = max22190_crc(&desc->buff[0]);
+
+	ret = no_os_spi_transfer(desc->comm_desc, &xfer, 1);
+	if (ret)
+		return ret;
+
+	if (desc->crc_en) {
+		crc = max22190_crc(&desc->buff[0]);
+		if (crc != (desc->buff[2] & 0x1F))
+			return -EINVAL;
+	}
+
+	*val = desc->buff[1];
+
+	return 0;
+}
+
+/**
+ * @brief Register write function for MAX22190
+ * @param desc - MAX22190 device descriptor.
+ * @param addr - Register value to which data is written.
+ * @param val - Value which is to be written to requested register.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int max22190_reg_write(struct max22190_desc *desc, uint32_t addr, uint32_t val)
+{
+	struct no_os_spi_msg xfer = {
+		.tx_buff = desc->buff,
+		.bytes_number = MAX22190_FRAME_SIZE,
+		.cs_change = 1,
+	};
+
+	if (desc->crc_en)
+		xfer.bytes_number++;
+
+	memset(desc->buff, 0, xfer.bytes_number);
+	desc->buff[0] = no_os_field_prep(MAX22190_ADDR_MASK, addr) |
+			no_os_field_prep(MAX22190_RW_MASK, 1);
+	desc->buff[1] = val;
+
+	if (desc->crc_en)
+		desc->buff[2] = max22190_crc(desc->buff);
+
+	return no_os_spi_transfer(desc->comm_desc, &xfer, 1);
+}
+
+/**
+ * @brief Register update function for MAX22190
+ * @param desc - MAX22190 device descriptor.
+ * @param addr - Register valueto wich data is updated.
+ * @param mask - Corresponding mask to the data that will be updated.
+ * @param val - Updated value to be written in the register at update.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int max22190_reg_update(struct max22190_desc *desc, uint32_t addr,
+			uint32_t mask, uint32_t val)
+{
+	int ret;
+	uint32_t reg_val = 0;
+
+	ret = max22190_reg_read(desc, addr, &reg_val);
+	if (ret)
+		return ret;
+
+	reg_val &= ~mask;
+	reg_val |= mask & val;
+
+	return max22190_reg_write(desc, addr, reg_val);
+}
+
+/**
+ * @brief MAX22190 Initialization function
+ * @param desc - MAX22190 device descriptor.
+ * @param param - MAX22190 initalization parameter.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int max22190_init(struct max22190_desc **desc,
+		  struct max22190_init_param *param)
+{
+	struct max22190_desc *descriptor;
+	int ret;
+
+	descriptor = no_os_calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	ret = no_os_spi_init(&descriptor->comm_desc, param->comm_param);
+	if (ret)
+		goto err;
+
+	descriptor->crc_en = param->crc_en;
+
+	ret = no_os_gpio_get_optional(&descriptor->en_gpio,
+				      param->en_gpio_param);
+	if (ret)
+		goto spi_err;
+
+	if (descriptor->en_gpio) {
+		ret = no_os_gpio_set_value(descriptor->en_gpio,
+					   NO_OS_GPIO_HIGH);
+		if (ret)
+			goto gpio_err;
+	}
+
+	ret = max22190_reg_update(descriptor, MAX22190_FAULT1_REG, NO_OS_BIT(6),
```

Comment:
What does this do? Adding a macro for the mask may help.

Reply:
A reg_update isn't really needed on that specific mask, so I will replace it with a simple register read to clear the latched faults.